dev/benchmarks/platform_channels_benchmarks/lib/main.dart

```diff
@@ -48,60 +48,116 @@ List<Object> _makeTestBuffer(int size) {
-    await channel.send(1234);
+  for (int i = 0; i < count; ++i) {
+    final List<Object> result =
+        await basicStandard.send(largeBuffer) as List<Object>;
+    // This check should be tiny compared to the actual channel send/receive.
+    size += (result == null) ? 0 : result.length;
   }
   watch.stop();
-  final double smallPayloadTime = watch.elapsedMicroseconds / numMessages;
 
-  watch.reset();
-  /// WARNING: Don't change the following line of code, it will invalidate
-  /// `Large` tests.  Instead make a different test.  The size of largeBuffer
-  /// serialized is 14214 bytes.
-  final List<Object> largeBuffer = _makeTestBuffer(1000);
+  if (size != largeBuffer.length * count) {
+    throw Exception(
+      'There is an error with the echo channel, the results don\'t add up: $size',
+    );
+  }
+
+  return watch.elapsedMicroseconds / count;
+}
 
+Future<double> _runBasicBinary(BasicMessageChannel<ByteData> basicBinary,
+    ByteData buffer, int count) async {
   int size = 0;
+  final Stopwatch watch = Stopwatch();
   watch.start();
-  for (int i = 0; i < numMessages; ++i) {
-    final List<Object> result = await channel.send(largeBuffer) as List<Object>;
+  for (int i = 0; i < count; ++i) {
+    final ByteData result = await basicBinary.send(buffer);
     // This check should be tiny compared to the actual channel send/receive.
-    size += (result == null) ? 0 : result.length;
+    size += (result == null) ? 0 : result.lengthInBytes;
   }
   watch.stop();
-  final double largePayloadTime = watch.elapsedMicroseconds / numMessages;
+  if (size != buffer.lengthInBytes * count) {
+    throw Exception(
+      'There is an error with the echo channel, the results don\'t add up: $size',
+    );
+  }
+
+  return watch.elapsedMicroseconds / count;
+}
 
-  if (size != largeBuffer.length * numMessages) {
+Future<void> _runTests() async {
+  if (kDebugMode) {
     throw Exception(
-        'There is an error with the echo channel, the results don\'t add up: $size');
+      "Must be run in profile mode! Use 'flutter run --profile'.",
+    );
   }
 
+  const BasicMessageChannel<Object> basicStandard = BasicMessageChannel<Object>(
+    'dev.flutter.echo.basic.standard',
+    StandardMessageCodec(),
+  );
+  const BasicMessageChannel<ByteData> basicBinary =
+      BasicMessageChannel<ByteData>(
+    'dev.flutter.echo.basic.binary',
+    BinaryCodec(),
+  );
+
+  /// WARNING: Don't change the following line of code, it will invalidate
+  /// `Large` tests.  Instead make a different test.  The size of largeBuffer
+  /// serialized is 14214 bytes.
+  final List<Object> largeBuffer = _makeTestBuffer(1000);
+  final ByteData largeBufferBytes =
+      const StandardMessageCodec().encodeMessage(largeBuffer);
+  final ByteData oneMB = ByteData(1024 * 1024);
+
+  const int numMessages = 2500;
+
   final BenchmarkResultPrinter printer = BenchmarkResultPrinter();
+  await _runBasicStandardSmall(basicStandard, 1);
```

Comment:
nit: here or in the method, consider adding ac omment about this being a warmup run.